### PR TITLE
 AIRFLOW-2107 add time_partitioning to run_query on BigQueryBaseCursor

### PR DESCRIPTION
**Pull request by marengaz**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-2107) issues and references them in the PR title. 

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
- added time_partitioning functionality to BigQueryBaseCursor.run_query
- found functionality already on BigQueryBaseCursor.run_load, so refactored to a function
- refactored tests accordingly too

### Tests
- [x] My PR has tests in TestTimePartitioningInRunJob 
- refactored to use mocking and new function

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by fenglu-db — COMMENTED** [review]
LGTM, thanks @marengaz

**Comment by criccomini** [comment]
LGTM as well, but the build failures need to be addressed

**Comment by criccomini** [comment]
(Might just require a rebase)

**Comment by marengaz** [comment]
how can i re-run tests? the failing one failed on a pip install

**Comment by Fokko** [comment]
@marengaz Wat works best for me, is a `git commit --amend` and then force push on the same branch.

**Comment by marengaz** [comment]
@criccomini - ok to merge before 1.10?

**Comment by kaxil** [comment]
LGTM

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -1485,3 +1490,17 @@ def var_print(var_name):
         project_id = default_project_id
 
     return project_id, dataset_id, table_id
+
+
+def _cleanse_time_partitioning(destination_dataset_table, time_partitioning_in):
+    # if it is a partitioned table ($ is in the table name) add partition load option
+    time_partitioning_out = {}
+    if destination_dataset_table and '$' in destination_dataset_table:
+        assert not time_partitioning_in.get('field'), (
```
Comment (kbl) [inline airflow/contrib/hooks/bigquery_hook.py:1499]: Why this assertion was placed there? When there is a table that is partitioned by custom column and I'd like to load results of some query to it it fails with one of two reasons:

1. when destination_dataset_table=table$partition is specified and time_partitioning is specified it fails on this assertion
2. when destination_dataset_table=table$partition is specified and time_partitioning is missing it fails with "'Incompatible table partitioning specification. Expects partitioning specification interval(type:day,field:time), but input partitioning specification is interval(type:day)'"

Comment (kaxil) [reply airflow/contrib/hooks/bigquery_hook.py:1499]: This has been resolved. This PR was done when BigQuery did not have functionality to partition by custom column

Comment (kaxil) [reply airflow/contrib/hooks/bigquery_hook.py:1499]: This was fixed in https://github.com/apache/incubator-airflow/pull/3901 and will be available in **Airflow 1.10.1**

Comment (kbl) [reply airflow/contrib/hooks/bigquery_hook.py:1499]: Thanks for an update, but I have one small note - I'm running airflow 1.10.0 on production, it was already released and that fix isn't there.

Comment (kaxil) [reply airflow/contrib/hooks/bigquery_hook.py:1499]: Sorry, silly me. I meant this will be available in **Airflow 1.10.1** (Next Release) . :)

**Pull request closed without merging**